kernel/api/src/main/java/org/sakaiproject/authz/api/AuthzGroupService.java

```diff
@@ -79,6 +79,14 @@
 	static final String AUTH_ROLE = ".auth";
 
 	/**
+	 * Get all provider IDs for the realms given.
+	 * 
+	 * @param realmIDs a List of the realms you want the provider IDs for.
+	 * @return a Map, where the key is the realm ID, and the value is a List of Strings of provider IDs for that realm
+	 */
+	public Map<String, List<String>> getProviderIDsForRealms( List<String> realmIDs );
```

Comment:
Sorry, I haven't taken a close look (but I wanted to mention this if you're looking at it again) but I think that you can just call `org.sakaiproject.site.impl.BaseSite#getProviderGroupId()`. This has the "advantage" that it's cached in the site so multiple on the same Site object won't go back to the DB. I haven't looked to see if `m_argz` is cached by EhCache.